src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -201,6 +201,19 @@ public function __construct($content = '', $status = 200, $headers = array())
         $this->setContent($content);
         $this->setStatusCode($status);
         $this->setProtocolVersion('1.0');
+
+        // Deprecations
+        $class = get_class($this);
+        if (__CLASS__ === $class || 0 === strpos($class, 'Symfony\\')) {
```

Comment:
Any vendor not beginning by `S` would pass the check. I'll try hard-coding a list.